docs/src/userguide/config/index.rst

```diff
@@ -144,6 +144,103 @@ access to configuration is needed for a transform's logic, then use the provided
             ...
 
 
+Templating and Overriding a Job's Configuration
+-----------------------------------------------
+
+Klio supports templating ``klio-job.yaml`` as well as overriding the configuration's values via the CLI.
+
+Templating
+^^^^^^^^^^
+
+To template any key or value in ``klio-job.yaml``, declare it with ``${...}``, for example:
+
+.. code-block:: yaml
+
+  # klio-job.yaml snippet
+  job_name: my-job
+  job_config:
+    a_key: ${my_template_value}
+    ${a_template_key}: some_static_value
+
+
+Then, use the ``--template`` flag (supported on most ``klio job`` subcommands – see :doc:`CLI documentation </reference/cli/api/cli/job>` for which commands supports the flag) to provide the content for the templated variables:
+
+.. code-block:: sh
+
+  # command accepts multiple `--template` flags
+  $ klio job run \
+    --template my_template_value=foo \
```

Comment:
Maybe someone else can verify, but I've only been able to get this to work when it's formatted as `--template=my_template_value=foo`, using `=` instead of a space.  Same for overrides or apparently any multi-value option.

Reply:
I actually did verify that this worked. In `klio_cli.cli.py`, I modified the `run` function like so:

```py
@job.command("run", help="Run a klio job.")
@core_options.image_tag(default=None, show_default="``git-sha[dirty?]``")
@options.runtime
@core_utils.with_klio_config
def run_job(klio_config, config_meta, **kwargs):
    import json
    print(json.dumps(klio_config.as_dict()))
    return
```

And then just ran
```sh
klio job run \
  --template my_template_value=foo \
  --template a_template_key=some_variable_key \
  | jq .job_config
```

and all looked good to me. I just ran it again with `--template=my_template_value=foo` and it's the same behavior.

Reply:
yep I just went back and double checked and it worked for me as well.  Not sure what I was doing when I initially had issues 🤷